### DISCUSSION TITLE
Integration external OAuth2 provider

### DISCUSSION THREAD

**Discussion by teddysbeach** [pocketbase/pocketbase#973] [Q&A] [answered]
Hi! I am trying to integrate OAuth2 into my Pocketbase project.
But the OAuth2 provider that I am trying to add is not on the list, 
so I need to ask how I can integrate the external OAuth2 provider.

(fyi, the Oauth2 provider is Kakao,  https://developers.kakao.com/docs/latest/en/kakaologin/common)

**Comment by ganigeorgiev** [comment, marked as answer]
Currently developers cannot register external OAuth2 providers due to various reasons (sometimes even the settings in the Admin UI may be different depending on the required provider configurations).

While you can't register Kakao as provider externally, it is possible to add the integration in the PocketBase source and make it available for everyone. Users contributions are welcomed. At a first glance looking at the docs, Kakao doesn't seem to support the PKCE flow. You still can integrate it with PocketBase but if you plan to use it on mobile you'll loose the added security when exchanging tokens. For a simple OAuth2 example integration, you could check the recent Spotify PR - https://github.com/pocketbase/pocketbase/pull/887/files (you can ignore the `ui/dist` files).
